test/uk/gov/hmrc/minorentityidentificationfrontend/services/SubmissionServiceSpec.scala

```diff
@@ -0,0 +1,107 @@
+
+        List(DetailsNotFound, DetailsMismatch).foreach(knownFactsMatchFailure => {
+
+          mockStorageService.retrieveUtr(testJourneyId) returns Future.successful(Some(Sautr(testSautr)))
+          mockStorageService.retrieveSaPostcode(testJourneyId) returns Future.successful(Some(testSaPostcode))
+          mockStorageService.retrieveCHRN(testJourneyId) returns Future.successful(Some(testCharityHMRCReferenceNumber))
+
+          mockValidateTrustKnownFactsService.validateTrustKnownFacts(journeyId = testJourneyId,
+            optSaUtr = Some(testSautr),
+            optSaPostcode = Some(testSaPostcode),
+            optCHRN = Some(testCharityHMRCReferenceNumber)) returns Future.successful(knownFactsMatchFailure)
+
+          val result = await(TestSubmissionService.submit(testJourneyId, testTrustJourneyConfig))
+
+          result mustBe errorRoutes.CannotConfirmBusinessController.show(testJourneyId).url
+
+          reset(mockStorageService, mockValidateTrustKnownFactsService)
+
+        })
+      }
+    }
+
+    "return continue url" when {
+      "match is SuccessfulMatch or match is UnMatchableWithoutRetry" in {
+
+        List(SuccessfulMatch, UnMatchableWithoutRetry).foreach(knownFactsMatchResult => {
+
+          mockStorageService.retrieveUtr(testJourneyId) returns Future.successful(Some(Sautr(testSautr)))
+          mockStorageService.retrieveSaPostcode(testJourneyId) returns Future.successful(Some(testSaPostcode))
+          mockStorageService.retrieveCHRN(testJourneyId) returns Future.successful(Some(testCharityHMRCReferenceNumber))
+
+          mockValidateTrustKnownFactsService.validateTrustKnownFacts(journeyId = testJourneyId,
+            optSaUtr = Some(testSautr),
+            optSaPostcode = Some(testSaPostcode),
+            optCHRN = Some(testCharityHMRCReferenceNumber)) returns Future.successful(knownFactsMatchResult)
+
+          val result = await(TestSubmissionService.submit(testJourneyId, testTrustJourneyConfig))
+
+          result mustBe testTrustJourneyConfig.continueUrl + s"?journeyId=$testJourneyId"
+
+          reset(mockStorageService, mockValidateTrustKnownFactsService)
+        })
+      }
+    }
+
+    "delegates to the ValidateTrustKnownFactsService any value" in {
+
+      List(Some("someValue"), None).foreach((someValue: Option[String]) => {
+
+        mockStorageService.retrieveUtr(testJourneyId) returns Future.successful(someValue.map(Sautr))
+        mockStorageService.retrieveSaPostcode(testJourneyId) returns Future.successful(someValue)
+        mockStorageService.retrieveCHRN(testJourneyId) returns Future.successful(someValue)
+
+        mockValidateTrustKnownFactsService.validateTrustKnownFacts(journeyId = testJourneyId,
+          optSaUtr = someValue,
+          optSaPostcode = someValue,
+          optCHRN = someValue) returns Future.successful(UnMatchableWithoutRetry)
+
+        await(TestSubmissionService.submit(testJourneyId, testTrustJourneyConfig))
+
```

Comment:
Not quite sure what this is doing. It doesn't assert anything in the test

Reply:
Just that delegates everything .... whatever values you passes is delegate them to the service. The previous tests does not prove for example that they delegate None value if:
storageService.retrieveUtr(journeyId)
storageService.retrieveSaPostcode(journeyId)
storageService.retrieveCHRN(journeyId)
retrieve None values
this tests proves it.
I can remove it if it is too confusing

Reply:
okay resolved during the chat